portfolio/src/main/webapp/script.js

```diff
@@ -16,8 +16,19 @@
  * Adds a random greeting to the page.
  */
 
-async function getHelloMessage(){
+async function getQuote(){
     const response = await fetch("/data");
-    const msg = await response.text();
-    document.getElementById("msg").innerText = msg;
+    const quote = await response.text();
+    document.getElementById("quote-container").innerText = quote;
+}
+async function getMessages(){
+    const list = document.getElementById("message-list");
+    const response = await fetch("/messages");
+    const messages = await response.json();
+    
+    messages.forEach(function(item){
+        const listElement = document.createElement("li");
+        listElement.innerHTML = item;
```

Comment:
Watch out for innerHTML! It's very easy to inadvertently introduce security vulnerabilities with it. (Especially as code/design evolve.) Prefer innerText/contentText 

See also: https://xss-game.appspot.com/